File: proselint/checks/misc/currency.py
"""
Currency.

---
layout:     post
source:     SublimeLinter-annotations
source_url: http://bit.ly/16Q7H41
title:      symbols
date:       2014-06-10 12:31:19
categories: writing
---

Currency symbols.

"""

from proselint.registry.checks import Check, types

check = Check(
    check_type=types.ExistenceSimple(
        pattern=r"\B\$[\d]* ?(?:dollars|usd|us dollars)\b",
    ),
    path="misc.currency",
    message="Incorrect use of symbols in {}.",
)

__register__ = (check,)

File: proselint-rs/crates/proselint-checks/src/misc/currency.rs
use proselint_registry::{pad, checks::{Check, types::*, Padding}};

pub const EXAMPLES_PASS: &[&str] = &["Smoke phrase with nothing flagged."];
pub const EXAMPLES_FAIL: &[&str] = &["It cost $10 dollars."];

const CHECK: Check = Check {
	check_type: &ExistenceSimple {
		pattern: pad!(
			Padding::SeparatorInText,
			r"\$[\d]* ?(?:dollars|usd|us dollars)"
		),
		exceptions: &[],
	},
	path: "misc.currency",
	msg: "Incorrect use of symbols in {}.",
	..Check::default()
};

pub const REGISTER: &[Check] = &[CHECK];
